lib/events.js

```diff
@@ -98,6 +98,56 @@ EventEmitter.prototype.getMaxListeners = function getMaxListeners() {
   return $getMaxListeners(this);
 };
 
+// Returns the longest sequence of `a` that fully appears in `b`,
+// of length at least 3.
+// This is a lazy approach but should work well enough, given that stack
+// frames are usually unequal or otherwise appear in groups, and that
+// we only run this code in case of an unhandled exception.
+function longestSeqContainedIn(a, b) {
+  for (var len = a.length; len >= 3; --len) {
+    for (var i = 0; i < a.length - len; ++i) {
+      // Attempt to find a[i:i+len] in b
+      for (var j = 0; j < b.length - len; ++j) {
+        let matches = true;
+        for (var k = 0; k < len; ++k) {
+          if (a[i + k] !== b[j + k]) {
+            matches = false;
+            break;
+          }
+        }
+        if (matches)
+          return [ len, i, j ];
+      }
+    }
+  }
+
+  return [ 0, 0, 0 ];
+}
+
+function longestCommonSubsequence(a, b) {
+  const [ l1, i1, j1 ] = longestSeqContainedIn(a, b);
+  const [ l2, i2, j2 ] = longestSeqContainedIn(b, a);
+  if (l1 > l2)
+    return [ l1, j1, i1 ];
+  else
+    return [ l2, i2, j2 ];
+}
```

Comment:
This part confuses me. As far as I see it the common part will always be identical or do I miss something?

Reply:
@BridgeAR Thanks for catching – it made sense in an earlier version of this code, but it’s removed now.